File: 2.0/java/webhooks/updateWebhook.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.webhooks.model.ResourceType;
import com.callfire.api.client.api.webhooks.model.Webhook;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api login", "api password");
        Webhook webhook = client.webhooksApi().get(11646003L);
        webhook.setName("sms sent update");
        webhook.getEvents().add(ResourceType.ResourceEvent.FINISHED);
        webhook.setCallback("https://callback-service.com/listener2");
        client.webhooksApi().update(webhook);
    }
}

File: 2.0/csharp/webhooks/updateWebhook.cs
using System.Collections.Generic;
using CallfireApiClient;
using CallfireApiClient.Api.Webhooks.Model;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var webhook = new Webhook
        {
            Id = 11646003,
            Name = "sms sent update",
            Resource = ResourceType.TEXT_BROADCAST,
            Events = new HashSet<ResourceEvent>
            {
                ResourceEvent.FINISHED
            },
            Callback = "https://callback-service.com/listener2"
        };
        client.WebhooksApi.Update(webhook);
    }
}
